### PR TITLE
Rename MSON Namespace to Data Structure Namespace

### PR DESCRIPTION
**Pull request by kylef**
This pull request implements [RFC-0001](https://github.com/refractproject/rfcs/blob/master/text/0001-mson-rename.md).

### CODE REVIEW

File: namespaces/resource-namespace.md [hunk 1]
```diff
@@ -49,15 +49,15 @@ The value of the `Templated Href` type SHOULD be resolved as a URI-Reference per
 
 ### Href Variables (Object Type)
 
-The definition is a MSON namespace `Object Type` where keys are respective URI Template variables.
+The definition is a Data Structures namespace `Object Type` where keys are respective URI Template variables.
```
Comment (pksunkara) [inline namespaces/resource-namespace.md]:
``` diff
- Data Structures
+ Data Structure
```

File: namespaces/resource-namespace.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ This document extends [Refract][] [MSON Namespace][] to define REST Resource dat
         - [Href (string)](#href-string)
         - [Templated Href (string)](#templated-href-string)
         - [Href Variables (Object Type)](#href-variables-object-type)
-        - [Data Structure (MSON Element)](#data-structure-mson-element)
+        - [Data Structure (Data Structure Element)](#data-structure-element)
```
Comment (pksunkara) [inline namespaces/resource-namespace.md]:
``` diff
- #data-structure-element
+ #data-structure-data-structure-element
```

File: namespaces/data-structure-namespace.md [hunk 3]
```diff
@@ -130,14 +128,14 @@ from the example above we get:
...
+In this namespace, every data structure is a sub-type of another data structure, and, therefore, it is directly or indirectly derived from one of the Data Structure _Base Types_. This is expressed as an inheritance of elements in Data Structure Refract. Where the predecessor of an element is referred to as element's _Base Element_.
 
-Note: Not every MSON _Base Type_ is presented in Refract primitive types and vice versa – see the table below.
+Note: Not every Data Structure _Base Type_ is presented in Refract primitive types and vice versa – see the table below.
 
 ### Type comparison
 
-| JSON primitive |      Refract     | MSON Base Type | MSON Namespace |
-|:--------------:|:----------------:|:--------------:|:--------------:|
+| JSON primitive |      Refract     | [MSON][MSON] Base Type | Data Structure Namespace |
```
Comment (pksunkara) [inline namespaces/data-structure-namespace.md]: Why is it `[MSON][MSON]`? Can't it be simply `[MSON][]`?

Comment (kylef) [inline namespaces/data-structure-namespace.md]: You're right, I forgot about this shorthand.

**Comment by pksunkara** [comment]
Finished review. Just one minor comment left.

**Pull request merged**